omod/src/main/java/org/openmrs/module/providermanagement/fragment/controller/PersonNameFragmentController.java

```diff
@@ -14,13 +14,50 @@
 
 package org.openmrs.module.providermanagement.fragment.controller;
 
-import org.openmrs.layout.web.name.NameSupport;
+import java.lang.reflect.Method;
+import java.lang.reflect.InvocationTargetException;
+
 import org.openmrs.ui.framework.fragment.FragmentModel;
 
 public class PersonNameFragmentController {
 
-    public void controller(FragmentModel model) {
-        model.addAttribute("layoutTemplate", NameSupport.getInstance().getDefaultLayoutTemplate());
-    }
+	/**
+	 * @param model
+	 * @throws ClassNotFoundException
+	 * @throws SecurityException
+	 * @throws NoSuchMethodException
+	 * @throws InvocationTargetException
+	 * @throws IllegalArgumentException
+	 * @throws IllegalAccessException
+	 */
+
+	public void controller(FragmentModel model) throws ClassNotFoundException, NoSuchMethodException, SecurityException,
+			IllegalAccessException, IllegalArgumentException, InvocationTargetException {
+
+
+				/*
+				 * backward compatibility
+				 */
+				Class<?> nameSurpport;
+
+				try {
+					nameSurpport = Class.forName("org.openmrs.layout.name.NameSupport");
+				} catch (ClassNotFoundException e) {
+					nameSurpport = Class.forName("org.openmrs.layout.web.name.NameSupport");
+				}
+
+				if (nameSurpport == null) {
+					return;
+				}
+
+				Method getInstance = nameSurpport.getDeclaredMethod("getInstance");
```

Comment:
@josephbate , you didn't have to change the Logic. you were initially getting this Method from the LayoutSupport class . This would probably throw an error
`Class<?> cls1 = Class.forName("org.openmrs.layout.LayoutSupport");`

Reply:
i tested it and it worked fine 
did you get any errors?
coz last time i was using `getDeclaredMethod()` instead of `getMthod()` so it worked when i run the sdk i had no problems 
so are you getting and failures on yourside?

Reply:
true , 
`getDeclaredMethod()` includes all methods declared by the class itself, whereas` getMethod()` returns only public methods, but also those inherited from a base class